### PR TITLE
Images routing with modals

### LINKED ISSUE #8350: Images routing with modals
Based on [this Slack conversation](https://wellcome.slack.com/archives/C8X9YKM5X/p1661434915131389)

1. Access `/images`
2. Click an image, opening a modal
3. Try to go back using the browser function
4. If URL is still `/image`, the modal stays open
5. And it's always the latest accessed image.

This behaviour should change so users could share specific URLs or access previous images they've clicked on as currently they "lose" them forever when browsing from one related image to the next.

### PR DESCRIPTION
**Pull request by rcantin-w**
## Who is this for?
Users

## What is it doing for them?
Allows them to browse through images, going back and forth when clicking related images. Currently it only keeps displaying the same image until the user leaves `/images`. Would also allow for sharing `/images` with a modal opened if the user wanted it.

Closes #8350

### CODE REVIEW

File: catalogue/webapp/components/VisuallySimilarImagesFromApi/VisuallySimilarImagesFromApi.tsx [hunk 1]
```diff
@@ -52,7 +52,7 @@ const VisuallySimilarImagesFromApi: FunctionComponent<Props> = ({
       <h3 className={font('wb', 5)}>Visually similar images</h3>
       <Wrapper>
         {similarImages.map(related => (
-          <a key={related.id} onClick={() => onClickImage(related)} href="#">
+          <a key={related.id} onClick={() => onClickImage(related)}>
```
Comment (gestchild) [inline catalogue/webapp/components/VisuallySimilarImagesFromApi/VisuallySimilarImagesFromApi.tsx:55]: I think removing the href, will remove the ability to tab to the link. Does make me question whether this should be a link - but that is perhaps one for another day.

Comment (rcantin-w) [reply catalogue/webapp/components/VisuallySimilarImagesFromApi/VisuallySimilarImagesFromApi.tsx]: You're right. I've pushed a new commit with a suggestion where it's still a link, but at least it follows the model I've done for the Results images, what do you think?

File: catalogue/webapp/components/ImageEndpointSearchResults/ImageEndpointSearchResults.tsx [hunk 2]
```diff
@@ -19,6 +23,52 @@ const ImageEndpointSearchResults: FunctionComponent<Props> = ({
...
+        {
+          shallow: true,
+        }
+      );
+    }
+  }, [isActive]);
+
+  useEffect(() => {
+    // If there is an imageId in the query, fetch that image and display it
+    // Otherwise ensure modal is closed and URL gets resetted
```
Comment (gestchild) [inline catalogue/webapp/components/ImageEndpointSearchResults/ImageEndpointSearchResults.tsx:68]: resetted to reset

File: catalogue/webapp/components/ImageEndpointSearchResults/ImageEndpointSearchResults.tsx [hunk 3]
```diff
@@ -19,6 +23,52 @@ const ImageEndpointSearchResults: FunctionComponent<Props> = ({
...
+    const image: Image = await fetch(apiUrl).then(res => res.json());
+
+    if (image) {
+      setExpandedImage(image);
+      setIsActive(true);
+    }
+  };
+
+  useEffect(() => {
+    // If there is a set expandedImage and it's a different one that the current queried one
```
Comment (gestchild) [inline catalogue/webapp/components/ImageEndpointSearchResults/ImageEndpointSearchResults.tsx:38]: that to than

**Review by gestchild — APPROVED** [review]
nice

File: catalogue/webapp/components/ImageEndpointSearchResults/ImageEndpointSearchResults.tsx (hunk 2 above)
Comment (rcantin-w) [reply catalogue/webapp/components/ImageEndpointSearchResults/ImageEndpointSearchResults.tsx:68]: thanks, my English tends to go down when ill 😄  I'll change both and merge!

**Pull request merged**